### ISSUE TITLE
Accessibility improvements for finding in files

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#69741]
Issue Type: <b>Bug</b>

CC: @isidorn

Hi,

Some parts of the workflow for searching within a file with ctrl+f aren't that accessible. As a demonstration, have a blank editor with the following contents:

```
hello world
goodbye world
```

From the top of the file, when pressing ctrl+f and typing `world`, visually the first occurance is highlighted, but no screen reader feedback is provided at all. So my instinct is to press enter, which of course brings me to the second match instead. 
Next when on the second line, perform the same search. By default, it wraps you back to the beginning of the file if no more matches are found, which is doubly confusing due to the lack of feedback.

I expect to hear something similar to how its announced in the results view in project-wide search, but less verbose. E.g, "1/2: Found 'world' at 9:20 with text: <contents of line>"

Similar context should also be provided when moving using f3 and shift+f3, but haven't figured out what the best messages would be.

On a related note, I'm also pondering whether disabling such wrap-around by default if a screen reader is detected would be better. It is obvious that this has happened visually, but unless enough context is provided, it can be confusing.

VS Code version: Code - Insiders 1.32.0-insider (804373ac4d5c84e96c8086cee5b9be132df88837, 2019-03-01T18:22:35.336Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.86GB (8.89GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|36%|

</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.25.0
vscode-eslint|dba|1.8.0
python|ms-|2019.2.5433
cpptools|ms-|0.21.0
Go|ms-|0.9.2
rust|rus|0.5.3
markdown-preview-enhanced|shd|0.3.11

</details>

**Comment by rebornix** [comment]
> I expect to hear something similar to how its announced in the results view in project-wide search, but less verbose. E.g, "1/2: Found 'world' at 9:20 with text: "

this one is a bug we should fix.

> I'm also pondering whether disabling such wrap-around by default if a screen reader is detected would be better

IMO We need to provide more hints for screen reader when we reach the last item in the document, and users can be aware that the next search will go back to the top of the document. To allow users disable this feature, we have a tracking item https://github.com/Microsoft/vscode/issues/28666 , we can introduce a setting if it doesn't make sense to some users.

**Comment by Neurrone** [comment]
@rebornix I think that having better alerts when focus moves when moving to the next/previous result would be sufficient; if I was at 3/3 and suddenly it says 1/3, then I know its moved.

Another possibility is to play the system bell sound every time focus is wrapped back to the top.

Regarding the proposed format of the alert, I think it can be condensed to "1/2: Found 'world' at line 9:20: <line contents>".

**Comment by isidorn** [comment]
@rebornix can you maybe provide a code pointer for this. So we could add a help-wanted, or I might look into this if I find time.
Thanks!

**Comment by rebornix** [comment]
@isidorn thanks for the ping. The code change is intuitive by leveraging `aria.status`.

Firstly, we listen to find results change in https://github.com/microsoft/vscode/blob/master/src/vs/editor/contrib/find/findWidget.ts#L324 and update labels when search results count change. So we can alert the status at the same position.

Secondly, for the message format, we can follow Search Viewlet ones but less verbose as suggested by @Neurrone .  https://github.com/microsoft/vscode/blob/579966f90d84df7186510e31b94fe464843cfbc9/src/vs/workbench/contrib/search/browser/searchView.ts#L1319-L1335

**Comment by isidorn** [comment]
@rebornix  Great. Thanks for the pointers. Let's see if some user jumps on this.
If not, I can take a look in the future.

**Comment by georgebatalinski** [comment]
@rebornix @isidorn I will have a look at this - 

```
I expect to hear something similar to how its announced in the results view in project-wide search, but less verbose. E.g, "1/2: Found 'world' at 9:20 with text: "
```

I tested with NVDA 2019.1.1 - and the above issue is reproducible

**Comment by isidorn** [comment]
@georgebatalinski thanks for being interested. Feel free to provide a PR. @rebornix provided nice code pointers.

**Comment by georgebatalinski** [comment]
@isidorn this is my first guess at the issue - let me know your thoughts

1. Re-using the existing `label` document.createTextNode(label) - would be the most beneficial 
- as hiding the text from the screen - has not worked for me - under `matchesCount` with NVDA
```
.sr-only {
  position: absolute;
  left: -10000px;
  width: 1px;
  height: 1px;
  overflow: hidden;
}
```

2. role="status" - is not consistent, hence I set it for `'role', 'alert'`


https://github.com/microsoft/vscode/pull/77031

**Comment by isidorn** [comment]
@georgebatalinski thanks! I will comment in the PR directly.

**Comment by rebornix** [comment]
The PR is already merged.

**Comment by Neurrone** [comment]
@georgebatalinski this is such a massive quality of life improvement and does exactly what I would expect, thanks a lot!
